react-delegationdashboard/src/components/Overview/Header/SetAgencyMetaDataModal.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState } from 'react';
+import { Modal } from 'react-bootstrap';
+import { useDelegation } from 'helpers';
+import { ErrorMessage, Formik } from 'formik';
+import { object, string } from 'yup';
+import { useContext } from 'context';
+import { AgencyMetadata } from 'helpers/types';
+
+const SetAgencyMetaDataModal = () => {
+  const { delegation } = useDelegation();
+  const { agencyMetaData } = useContext();
+  const [showDelegateModal, setShowDelegateModal] = useState(false);
+  const handleReDelegationCapActivation = (values: AgencyMetadata) => {
+    const hexName = Buffer.from(values.name).toString('hex');
+    const hexWeb = Buffer.from(values.website).toString('hex');
+    const hexKeyBase = Buffer.from(values.keybase).toString('hex');
+    const data = hexName + '@' + hexWeb + '@' + hexKeyBase;
+    delegation.sendTransaction('0', 'setMetaData', data).then();
+  };
+
+  return (
+    <>
+      <button
+        onClick={() => {
+          setShowDelegateModal(true);
+        }}
+        className="btn btn-primary btn-sm ml-3"
+      >
+        Identity
+      </button>
+      <Modal
+        show={showDelegateModal}
+        onHide={() => setShowDelegateModal(false)}
+        className="modal-container"
+        animation={false}
+        centered
+      >
+        <div className="card">
+          <div className="card-body p-spacer text-center">
+            <p className="h6 mb-spacer" data-testid="metaDataTitle">
+              Agency Details
+            </p>
+            <p className="mb-spacer">
+              Update or set your agency details in order to validate your identity.
+            </p>
+            <Formik
+              initialValues={agencyMetaData}
+              onSubmit={values => {
+                handleReDelegationCapActivation(values);
+              }}
+              validationSchema={object().shape({
+                website: string()
+                  .required('Required')
+                  .test('URL', 'URL is not valid!', value => {
```

Comment:
can you remove the validation for https, it is ok to let the user only add the site without the https://

Reply:
I would suggest using https://developer.mozilla.org/en-US/docs/Web/API/URL/URL. If `new URL(value)` fails then false, else true

Reply:
```new URL()``` is still asking to use ```https://``` or ```http://```

Reply:
I added a new regex that handles all cases ```www.test.com/test.com/http(s)://test.com/http(s)://www.test.com```